internal/federationout/federationout.go

```diff
@@ -86,153 +87,253 @@ func (s Server) Fetch(ctx context.Context, req *pb.FederationFetchRequest) (*pb.
 	return response, nil
 }
 
-func (s Server) fetch(ctx context.Context, req *pb.FederationFetchRequest, itFunc iterateExposuresFunc, fetchUntil time.Time) (*pb.FederationFetchResponse, error) {
+func (s Server) fetch(ctx context.Context, req *federation.FederationFetchRequest, itFunc iterateExposuresFunc, fetchUntil time.Time) (*federation.FederationFetchResponse, error) {
 	logger := logging.FromContext(ctx)
 	metrics := s.env.MetricsExporter(ctx)
 	metricsMiddleWare := metricsware.NewMiddleWare(&metrics)
 
-	for i := range req.RegionIdentifiers {
-		req.RegionIdentifiers[i] = strings.ToUpper(req.RegionIdentifiers[i])
+	if in := intersect(req.IncludeRegions, req.ExcludeRegions); len(in) > 0 {
+		logger.Errorw("overlap in included and excluded regions", "intersection", in)
+		return nil, fmt.Errorf("overlap in include and exclude region set: %v", in)
 	}
-	for i := range req.ExcludeRegionIdentifiers {
-		req.ExcludeRegionIdentifiers[i] = strings.ToUpper(req.ExcludeRegionIdentifiers[i])
+
+	for i, region := range req.IncludeRegions {
+		req.IncludeRegions[i] = strings.ToUpper(region)
+	}
+	for i, exRegion := range req.ExcludeRegions {
+		req.ExcludeRegions[i] = strings.ToUpper(exRegion)
+	}
+	metricsMiddleWare.RecordFetchRegionsRequested(ctx, len(req.IncludeRegions))
+	metricsMiddleWare.RecordFetchRegionsExcluded(ctx, len(req.ExcludeRegions))
+
+	// Use configuration max or user provided max.
+	maxRecords := s.config.MaxRecords
+	if req.MaxExposureKeys > 0 && req.MaxExposureKeys < maxRecords {
```

Comment:
Should there be some kind or log or error returned if the request exceeds the max?